sdk/appconfiguration/app-configuration/src/internal/helpers.ts

```diff
@@ -311,3 +308,19 @@ export function errorMessageForUnexpectedSetting(
 ): string {
   return `Setting with key ${key} is not a valid ${expectedType}, make sure to have the correct content-type and a valid non-null value.`;
 }
+
+export function assertResponse<T extends object>(
+  result: T
+): asserts result is T & HttpResponseField<any> {
+  if (!hasUnderscoreResponse(result)) {
+    throw new Error(
```

Comment:
I wonder if we should instead log this (and maybe have it fail our tests as well) -- basically I would be a little concerned about breaking someone's usage of app-config if they don't rely on _response. Maybe if it doesn't exist we can attach a message on _response that says to file a bug? :)

 We can take this as a follow-up issue, though

Reply:
oh that's a great point.

Reply:
![image](https://user-images.githubusercontent.com/10452642/162277685-72e30a51-97db-4604-a4ca-b12e2008b733.png)
This good?

Reply:
Works for me!